classes/output/report_summary.php

```diff
@@ -126,6 +126,70 @@ private function calculate_total_cost() {
         return $total;
     }
 
+    /**
+     * Get the file summary totals from the DB.
+     * Used in generating chart data.
+     *
+     * @return array $totals The array of totals.
+     */
+    private function get_file_summary_totals() : array {
+        global $DB;
+        $totals = array();
+        $totalfiles = 0;
+        $videofiles = 0;
+        $audiofiles = 0;
+        $otherfiles = 0;
+
+        // Get values for chart from the database.
+        list($insql, $inparams) = $DB->get_in_or_equal(array('totalfiles', 'videofiles', 'audiofiles'));
+        $select = "name $insql";  // Don't count files added by smartmedia itself.
+        $values = $DB->get_records_select('local_smartmedia_reports', $select, $inparams, '', 'name, value');
+
+        if (!empty(($values))) { // Handle case where there is no data in table
+            $totalfiles = $values['totalfiles']->value;
+            $videofiles = $values['videofiles']->value;
+            $audiofiles = $values['audiofiles']->value;
+
+            $otherfiles = $totalfiles - ($videofiles + $audiofiles);
+            $totals = array($otherfiles, $videofiles, $audiofiles);
+        }
+
+        return $totals;
```

Comment:
Minor thing, but won't this error when there are no files, as `$total` is never initialised?

Reply:
And by error I mean, return `null`.

Reply:
$totals is initialised as an empty array on line 137 as a guard against this condition